spring-cloud-gcp-autoconfigure/src/main/java/com/google/cloud/spring/autoconfigure/pubsub/GcpPubSubAutoConfiguration.java

```diff
@@ -129,23 +134,6 @@ public ExecutorProvider publisherExecutorProvider(
 		return FixedExecutorProvider.create(scheduler.getScheduledExecutor());
 	}
 
-	@Bean
-	@ConditionalOnMissingBean(name = "pubsubSubscriberThreadPool")
-	public ThreadPoolTaskScheduler pubsubSubscriberThreadPool() {
```

Comment:
Why is it that we no longer need these beans? Couldn't they be used when there is no subscription-specific config?

Reply:
Good question. We made this bean optional for the subscriber factory in order to avoid breaking the current functionality for users who are using custom beans instead of autoconfiguration to set properties. If a custom bean is present then we use that otherwise we use the autoconfigured values. We need to remove our bean in order to prevent it from always getting injected into our subscriber factory and as a result, overriding the subscription-specific/default properties. Additionally, when there are no subscription-specific properties, `PubSubConfiguration` already takes care of using global (if specified) or default configurations as a fallback.